### PR TITLE
MS5611 failed to work on RPi on I2C bus

### PR DESCRIPTION
**Pull request by SalimTerryLi**
**Describe problem solved by this pull request**
Make MS5611 available on RPi with IIC bus.
MS5611 on /dev/i2c-1
start-up command: `ms5611 start -I -T 5611`

### CODE REVIEW

**Comment by SalimTerryLi** [comment]
@dagar

**Comment by dagar** [comment]
Nice work @SalimTerryLi. 

Are the white space changes intentional (https://github.com/PX4/Firmware/pull/13814/files)?

File: src/drivers/barometer/ms5611/ms5611_main.cpp [hunk 1]
```diff
@@ -105,15 +105,22 @@ start_bus(struct ms5611_bus_option &bus, enum MS56XX_DEVICE_TYPES device_type)
...
 	bus.dev = new MS5611(interface, prom_buf, bus.devpath, device_type);
 
-	if (bus.dev != nullptr && OK != bus.dev->init()) {
-		delete bus.dev;
-		bus.dev = NULL;
+	if (bus.dev != nullptr) {
+		if (bus.dev->init() != OK) {
+			PX4_ERR("MS5611 start failed");
+			delete bus.dev;
+			bus.dev = NULL;
```
Comment (dagar) [inline src/drivers/barometer/ms5611/ms5611_main.cpp]: In c++ use nullptr in place of NULL.

Comment (dagar) [inline src/drivers/barometer/ms5611/ms5611_main.cpp]:
```suggestion
			bus.dev = nullptr;
```

File: src/drivers/barometer/ms5611/ms5611_i2c.cpp [hunk 2]
```diff
@@ -253,3 +256,15 @@ MS5611_I2C::_read_prom()
...
+int MS5611_I2C::init()
+{
+	int ret = I2C::init();
+
+	if (ret != PX4_OK) {
+		PX4_ERR("I2C::init failed");
+		return ret;
+	}
+
+	return probe();
```
Comment (dagar) [inline src/drivers/barometer/ms5611/ms5611_i2c.cpp]: Actually, this is can and should be fixed lower level. In the NuttX version of the I2C wrapper `I2C::init()` calls probe. https://github.com/PX4/Firmware/blob/cc7a0cd69dec52db7816a24e58e782a82a8243a3/src/lib/drivers/device/nuttx/I2C.cpp#L140

I'll bring the posix version into sync. https://github.com/PX4/Firmware/blob/cc7a0cd69dec52db7816a24e58e782a82a8243a3/src/lib/drivers/device/posix/I2C.cpp#L74

**Comment by SalimTerryLi** [comment]
> Nice work @SalimTerryLi.
> 
> Are the white space changes intentional (https://github.com/PX4/Firmware/pull/13814/files)?

I just did 'Reformat Code' in Clion...

**Comment by dagar** [comment]
> > Nice work @SalimTerryLi.
> > Are the white space changes intentional (https://github.com/PX4/Firmware/pull/13814/files)?
> 
> I just did 'Reformat Code' in Clion...

We use astyle with this configuration. https://github.com/PX4/Firmware/blob/master/Tools/astyle/astylerc

If it didn't already exist I would have gone with clang-format, but it's hard to transition from one tool to another without thousands of minor changes.

With astyle installed locally you can run `make check_format` or `make format` to use the in tree config.

File: src/drivers/barometer/ms5611/ms5611_i2c.cpp (hunk 2 above)
Comment (dagar) [reply src/drivers/barometer/ms5611/ms5611_i2c.cpp]: https://github.com/PX4/Firmware/pull/13818

**Comment by SalimTerryLi** [comment]
> > > Nice work @SalimTerryLi.
> > > Are the white space changes intentional (https://github.com/PX4/Firmware/pull/13814/files)?
> > 
> > 
> > I just did 'Reformat Code' in Clion...
> 
> We use astyle with this configuration. https://github.com/PX4/Firmware/blob/master/Tools/astyle/astylerc
> 
> If it didn't already exist I would have gone with clang-format, but it's hard to transition from one tool to another without thousands of minor changes.
> 
> With astyle installed locally you can run `make check_format` or `make format` to use the in tree config.

I did 'Reformat code' before `make format`, Then I created the commit. Astyle may not deal with white spaces?

File: src/drivers/barometer/ms5611/ms5611_i2c.cpp (hunk 2 above)
Comment (dagar) [reply src/drivers/barometer/ms5611/ms5611_i2c.cpp]: Can you delete this now and retest? https://github.com/PX4/Firmware/pull/13818 has merged.

**Comment by dagar** [comment]
Confirmed ms5611 is still working on NuttX.

image

**Pull request merged**